composer/algorithms/cutmix/README.md

```diff
@@ -0,0 +1,92 @@
+# 🎃 CutMix
+
+[\[How to Use\]](#how-to-use) - [\[Suggested Hyperparameters\]](#suggested-hyperparameters) - [\[Technical Details\]](#technical-details) - [\[Attribution\]](#attribution)
+
+`Computer Vision`
+
+CutMix is a data augmentation technique that modifies images by cutting out a small patch and replacing it with a different image.
+It is a regularization technique that improves the generalization accuracy of models for computer vision.
+
+| ![CutMix](cutmix.png) |
+|:--:
+|*An image with CutMix applied. A picture of a cat has been placed over the top left corner of a picture of a dog. This is an image of [Figure 1 from Yun et al. (2019)](https://arxiv.org/abs/1905.04899).*|
+
+## How to Use
+
+### Functional Interface
+
+TODO(CORY): FIX
+
+```python
+def training_loop(model, train_loader):
+  opt = torch.optim.Adam(model.parameters())
+  loss_fn = F.cross_entropy
+  model.train()
+  
+  for epoch in range(num_epochs):
+      for X, y in train_loader:
+          y_hat = model(X)
+          loss = loss_fn(y_hat, y)
+          loss.backward()
+          opt.step()
+          opt.zero_grad()
+```
+
+### Composer Trainer
+
+TODO(CORY): Fix and provide commentary and/or comments
+
+```python
+from composer.algorithms import XXX
+from composer.trainer import Trainer
+
+trainer = Trainer(model=model,
+                  train_dataloader=train_dataloader,
+                  max_duration='1ep',
+                  algorithms=[
+                  ])
+
+trainer.fit()
+```
+
+### Implementation Details
+
+TODO(CORY): Briefly describe how this is implemented under the hood in Composer. Need to explain the mask in particular, since it shows up a few times below.
+
+## Suggested Hyperparameters
+
+Setting `alpha=1` is a standard choice. This produces a uniform distribution, meaning `lambda` (the interpolation between the labels of the two sets of examples) is selected uniformly between 0 and 1.
+
+## Technical Details
+
+CutMix works by creating a new mini-batch of inputs to the network by operating on a batch `(X1, y1)` of (inputs, targets) together with version `(X2, y2)` with the same examples but where the ordering of examples has been shuffled.
+The final set of inputs `X` is created by choosing a rectangular box within each example `x1` in `X1` and filling it with the data from the same region from the corresponding example `x2` in `X2`.
+The final set of targets `y` is created by sampling a value `lambda` (between 0.0 and 1.0) from the Beta distribution parameterized by `alpha` and interpolating between the targets `y1` and `y2`.
```

Comment:
@coryMosaicML 

Can you also include some commentary on the `adjusted_lambda` and how it's computed. Also, looking at the `_adjust_lambda` code in `cutmix.py`, it seems like the `cutmix_lambda` is not even used in computing the `adjusted_lambda` and that the interpolation is purely in terms of the `adjusted_lambda` not the `cutmix_lambda`. 

^Is this intended? If so, it's definitely worth noting this behavior.